### PR TITLE
Scroll chat to search result

### LINKED ISSUE #602: scroll chat to search result
the new global search also searches for messages, however, taping on a message-search-result, currently always shows the chat at the most recent position.

it would be great, if the chat scrolls to the message from the result directly.

might be, that we need a "real virtual" list of messages to target this issue (as just done for the contact list).

### PR DESCRIPTION
**Pull request by nayooti**
fixes #602

### CODE REVIEW

**Comment by cyBerta** [comment]
you could squash both commits to one, since the first one is basically reverted with the second commit, but if it's too cumbersome then nevermind :)

File: deltachat-ios/Controller/ChatViewController.swift [hunk 1]
```diff
@@ -381,6 +390,17 @@ class ChatViewController: MessagesViewController {
...
 
+    func scrollToMessage(id: Int, animated: Bool) {
+        for (index, msg) in messageList.enumerated() {
+            if msg.id == id {
+                let indexPath = IndexPath(row: 0, section: index)
+                messagesCollectionView.scrollToItem(at: indexPath, at: .top, animated: animated)
+                return
+            }
+        }
+        safe_fatalError("message could not be found")
```
Comment (cyBerta) [inline deltachat-ios/Controller/ChatViewController.swift]: this is too harsh. On my first try, the app already crashed. The reason was that the message was not within this first ones that are loaded, but already 'hidden'. I would have to pull to refresh in the chat view to have the searched message in the list.

Comment (cyBerta) [reply deltachat-ios/Controller/ChatViewController.swift]: Ok, this will not crash in production but still it's annoying a bit.

Comment (r10s) [reply deltachat-ios/Controller/ChatViewController.swift]: doesn't that always happen when clicking on a search result that is not in the last 100 or so messages?

**Comment by r10s** [comment]
> It would be super nice if the scroll feature would also work for older messages

this is why i pointed out that we may need '"real virtual" list of messages to target' at #602 - but yes, as a first step, this is already great.

File: deltachat-ios/Controller/ChatViewController.swift (hunk 1 above)
Comment (nayooti) [reply deltachat-ios/Controller/ChatViewController.swift]: Ok, I probably got this wrong. I didn't know/considered, that messages are incrementally fetched

**Comment by r10s** [comment]
closing this for now, i think sometimes scrolling to a messages an sometimes not is worse than just always showing the chat at the end.  
we can come back to that when we actually have a virtual message list and can scroll to random positions.

**Pull request closed without merging**